src/test/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBOTest.java

```diff
@@ -0,0 +1,112 @@
+package pl.pszczolkowski.mustdo.domain.task.bo;
+
+import javax.transaction.Transactional;
+
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.transaction.TransactionConfiguration;
+import org.springframework.test.context.web.WebAppConfiguration;
+
+import com.tngtech.jgiven.annotation.ProvidedScenarioState;
+import com.tngtech.jgiven.junit.ScenarioTest;
+
+import pl.pszczolkowski.mustdo.Application;
+import pl.pszczolkowski.mustdo.domain.board.bo.BoardBO;
+import pl.pszczolkowski.mustdo.domain.board.repository.BoardRepository;
+import pl.pszczolkowski.mustdo.domain.task.bo.steps.GivenTasksListBO;
+import pl.pszczolkowski.mustdo.domain.task.bo.steps.ThenTasksListBO;
+import pl.pszczolkowski.mustdo.domain.task.bo.steps.WhenTasksListBO;
+import pl.pszczolkowski.mustdo.domain.task.finder.TasksListSnapshotFinder;
+import pl.pszczolkowski.mustdo.domain.task.repository.TasksListRepository;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = Application.class)
+@WebAppConfiguration
+@Transactional
+@TransactionConfiguration(defaultRollback = true)
+public class TasksListBOTest extends ScenarioTest<GivenTasksListBO, WhenTasksListBO, ThenTasksListBO> {
+	private static final String CLAZZ = BoardBO.class.getSimpleName();
+
+	@Autowired
+	@ProvidedScenarioState
+	private BoardBO boardBO;
+	@Autowired
+	@ProvidedScenarioState
+	private TasksListBO tasksListBo;
+	@Autowired
+	@ProvidedScenarioState
+	private BoardRepository boardRepository;
+	@Autowired
+	@ProvidedScenarioState
+	private TasksListRepository tasksListRepository;
+	@Autowired
+	@ProvidedScenarioState
+	private TasksListSnapshotFinder tasksListSnapshotFinder;
+	
+	@After
+	public void tearDown(){
+		tasksListRepository.deleteAll();
+		boardRepository.deleteAll();
+	}
+	
+	@Test
+	public void should_add_new_tasksList(){
+		given().a_board()
```

Comment:
This is test case for task domain, don't create entities from other domains. Change it in other tests also